client/src/sensemakerStore.ts

```diff
@@ -1,29 +1,69 @@
-import { AgentPubKey, encodeHashToBase64, EntryHash, EntryHashB64, Record as HolochainRecord } from '@holochain/client';
-import { SensemakerService } from './sensemakerService';
+import { AgentPubKey, decodeHashFromBase64, encodeHashToBase64, EntryHash, EntryHashB64, Record as HolochainRecord } from '@holochain/client';
+import { derived, writable, Writable } from 'svelte/store';
+import { rxWritable } from 'svelte-fuse-rx';
+import { merge, filter, map, Observable } from 'rxjs';
+import { produce } from 'immer';
+import { createContext } from '@lit-labs/context';
+
+import { SensemakerService, ResourceAssessmentsResponse } from './sensemakerService';
 import { AppletConfig, AppletConfigInput, AppletUIConfig, Assessment, ComputeContextInput, CreateAppletConfigInput, CreateAssessmentInput, CulturalContext, Dimension, DimensionEh, GetAssessmentsForResourceInput, Method, ResourceDef, ResourceDefEh, ResourceEh, RunMethodInput } from './index';
-import { derived, get, Writable, writable } from 'svelte/store';
-import { EntryHashMap } from '@holochain-open-dev/utils'
 import { Option } from './utils';
-import { createContext } from '@lit-labs/context';
 
-interface ContextResults {
+// zome API output types
+
+export interface ContextResults {
   [culturalContextName: string]: EntryHash[],
 }
+
+export type ResourceAssessmentResults = Map<EntryHashB64, Set<Assessment>>
+
+// SensemakerStore API inputs
+
+export interface assessmentsFilterOpts {
+  resourceEhs: string[]
+  dimensionEhs: string[]
+}
+
+// TypeScript interface
+
+type AssessmentObservable = Observable<Set<Assessment>> & Writable<Set<Assessment>>
```

Comment:
I'm not super familiar with TS intersection types, could you elaborate on why we need to do this?

Reply:
I think `Writable` is a superclass of `Readable`. But we should add some unit tests around readables specifically and ensure everything works as expected.

Reply:
It is a superclass, but my nit-picky brain thinks that if you properly want to reason about the functors between structural types, you need to know the exact minimum interface necessary for each use of it. So do we need to ensure that something passed in has the `Readable` (`subscribe`) interface, or do we need something that is also `Writable` (with `set` and `update`)?